agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -62,17 +67,63 @@ def on_container_event(topic, e)
       if attrs['io.kontena.service.id'] == @service_pod.service_id &&
           attrs['io.kontena.service.instance_number'].to_i == @service_pod.instance_number
         @container_state_changed = true
+        handle_restart_on_die if DIE_EVENTS.include?(e.status)
+      end
+    end
+
+    # Handles events when container has died
+    def handle_restart_on_die
+      unless @service_pod.running?
+        @restart_backoff_timer.cancel if @restart_backoff_timer
+        return
+      end
+
+      return if apply_in_progress?
+
+      # cancel counters because we are called again
+      cancel_restart_timers
+
+      # backoff restarts
+      backoff = @restarts ** 2
+      backoff = MAX_RESTART_BACKOFF if backoff > MAX_RESTART_BACKOFF
```

Comment:
The max time is actually limited by the server sync-loop interval. And as the server sync-loop essentially also restarts died container, it makes the whole backoff period quite un-predictable. Would it be possible to make the sync-loop to respect the backoff policy? I.e. in the sync-loop "skip" starting the container if there's existing restart loop ongoing.